framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/deployment/DatabaseDeploymentImpl.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2017 JBoss by Red Hat.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kie.cloud.openshift.deployment;
+
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+import io.fabric8.kubernetes.api.model.Pod;
+import static java.util.stream.Collectors.toList;
+import org.kie.cloud.api.deployment.DatabaseDeployment;
+import org.kie.cloud.api.deployment.DatabaseInstance;
+import org.kie.cloud.api.deployment.Instance;
+import org.kie.cloud.openshift.OpenShiftController;
+import org.kie.cloud.openshift.resource.OpenShiftResourceConstants;
+
+public class DatabaseDeploymentImpl implements DatabaseDeployment {
+
+    private OpenShiftController openShiftController;
+    private String username;
+    private String password;
+    private String namespace;
+    private String database;
+    private URL url;
+    private String serviceName;
+
+    public void setOpenShiftController(OpenShiftController openShiftController) {
+        this.openShiftController = openShiftController;
+    }
+
+    public void setUsername(String username) {
+        this.username = username;
+    }
+
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+    public void setNamespace(String namespace) {
+        this.namespace = namespace;
+    }
+
+    public void setDatabaseName(String database) {
+        this.database = database;
+    }
+
+    public void setUrl(URL url) {
+        this.url = url;
+    }
+
+    public OpenShiftController getOpenShiftController() {
+        return openShiftController;
+    }
+
+    @Override
+    public URL getUrl() {
+        return url;
+    }
+
+    @Override
+    public String getUsername() {
+        return username;
+    }
+
+    @Override
+    public String getPassword() {
+        return password;
+    }
+
+    @Override
+    public String getDatabaseName() {
+        return database;
+    }
+
+    @Override
+    public String getNamespace() {
+        return namespace;
+    }
+
+    public String getServiceName() {
+        return serviceName;
+    }
+
+    public void setServiceName(String applicationName) {
+        this.serviceName = applicationName + "-" + database;
```

Comment:
This will require "database" property already being available. Maybe it would be better to move service name completion to getServiceName() and change this method to setApplicationName()